docs/site/Loopback-component-authentication.md

```diff
@@ -0,0 +1,603 @@
+`Authentication Component` takes care of the rest.
+
+## Authentication Component
+
+To utilize `authentication` in an application `application.ts`, you must load
+the authentication component named `AuthenticationComponent`.
+
+```ts
+export class MyApplication extends BootMixin(
+  ServiceMixin(RepositoryMixin(RestApplication)),
+) {
+  constructor(options?: ApplicationConfig) {
+    super(options);
+
+    //...
+
+    this.component(AuthenticationComponent);
+
+    //...
+  }
+}
+```
+
+The `AuthenticationComponent` is defined as follows:
+
+```ts
+export class AuthenticationComponent implements Component {
+  providers?: ProviderMap;
+
+  constructor() {
+    this.providers = {
+      [AuthenticationBindings.AUTH_ACTION.key]: AuthenticateActionProvider,
+      [AuthenticationBindings.STRATEGY.key]: AuthenticationStrategyProvider,
+      [AuthenticationBindings.METADATA.key]: AuthMetadataProvider,
+    };
+  }
+}
+```
+
+As you can see, there are a few
+[providers](https://loopback.io/doc/en/lb4/Creating-components.html#providers)
+which make up the bulk of the authenticaton component.
+
+Essentially
+
+- The binding key `AuthenticationBindings.AUTH_ACTION.key` is bound to
+  `AuthenticateActionProvider` which returns an authenticating function of type
+  `AuthenticateFn`
+- The binding key `AuthenticationBindings.STRATEGY.key` is bound to
+  `AuthenticationStrategyProvider` which resolves and returns an authenticating
+  strategy of type `AuthenticationStrategy`
+- The binding key `AuthenticationBindings.METADATA.key` is bound to
+  `AuthMetadataProvider` which returns authentication decorator metadata of type
+  `AuthenticationMetadata`
+
+The purpose of these providers and the values they return will be explained in
+the sections below.
+
+## Authentication Strategies
+
+Support for multiple authentication strategies is possible with a **common**
+authentication strategy interface, and an **extensionPoint/extensions** pattern
+used to **register** and **discover** authentication strategies.
+
+### AuthenticationStrategy Interface
+
+The `AuthenticationComponent` declares a common authentication strategy
+interface named `AuthenticationStrategy`.
+
+```ts
+export interface AuthenticationStrategy {
+  /**
+   * The 'name' property is a unique identifier for the
+   * authentication strategy ( for example : 'basic', 'jwt', etc)
+   */
+  name: string;
+
+  /**
+   * The 'authenticate' method takes in a given request and returns a user profile
+   * which is an instance of 'UserProfile'.
+   * (A user profile is a minimal subset of a user object)
+   * If the user credentials are valid, this method should return a 'UserProfile' instance.
+   * If the user credentials are invalid, this method should throw an error
+   * If the user credentials are missing, this method should throw an error, or return 'undefined'
+   * and let the authentication action deal with it.
+   *
+   * @param request
```

Comment:
Add some description for `request`, such as:

`@param request - Express request object`